zebra/zebra_rib.c

```diff
@@ -1175,10 +1175,7 @@ static void rib_uninstall(struct route_node *rn, struct route_entry *re)
 		if (zebra_rib_labeled_unicast(re))
 			zebra_mpls_lsp_uninstall(info->zvrf, rn, re);
 
-		if (!RIB_SYSTEM_ROUTE(re))
-			rib_uninstall_kernel(rn, re);
-		else
-			UNSET_FLAG(re->status, ROUTE_ENTRY_INSTALLED);
+		rib_uninstall_kernel(rn, re);
```

Comment:
what's the motivation for driving system route change through the dataplane system - that seems like quite a change? is there some problem that's being resolved?

Reply:
we had multiple code paths and checks that made the code quite complicated.  Sending through the data path reduces us to one code path and greatly simplifies the code path.  Which is why I made this change.